src/wasm.rs

```diff
@@ -70,6 +70,25 @@ impl Instant {
     pub fn elapsed(&self) -> Duration {
         Instant::now() - *self
     }
+
+    pub fn checked_duration_since(&self, earlier: Instant) -> Option<Duration> {
+        match self.cmp(&earlier) {
+            Ordering::Less => None,
+            _ => Some(self.duration_since(earlier)),
+        }
+    }
+
+    pub fn saturating_duration_since(&self, earlier: Instant) -> Duration {
+        self.checked_duration_since(earlier).unwrap_or_default()
+    }
+
+    pub fn checked_add(&self, duration: Duration) -> Option<Instant> {
+        Some(*self + duration)
+    }
+
+    pub fn checked_sub(&self, duration: Duration) -> Option<Instant> {
+        Some(*self - duration)
```

Comment:
This would allow creating a negative `Instance`, right?

```rust
Instant::now() - (Instant::now() + Duration::from_secs(1))
```

Reply:
It seems, I don't know for sure if it should be avoided or no

Reply:
I very much think this should be avoided (across the whole crate). I do think we should mimic the behavior of the standard library here.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6e4ac52feea310c855e5a4d420e2d8ea

Reply:
This highly depends of the underlying implementation and could be different on an other platform. Nevertheless we could restrict the use of negative Instant.

I can't easily edit the PR for now, but I'll be back on August.